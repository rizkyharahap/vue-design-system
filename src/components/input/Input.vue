<script setup lang="ts">
import { baseClass, type InputPropsProps, inputSizes, inputVariants } from ".";

withDefaults(defineProps<InputPropsProps>(), {
  variant: "default",
  size: "normal",
});
</script>

<template>
  <input
    type="text"
    id="large-input"
    :class="` ${inputSizes[size]} ${inputVariants[variant]} ${baseClass}`"
  />
</template>
