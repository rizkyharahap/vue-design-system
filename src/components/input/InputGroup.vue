<script setup lang="ts">
import { type InputProps } from ".";

withDefaults(defineProps<InputProps>(), {
  variant: "default",
  size: "normal",
});
</script>

<template>
  <div class="relative">
    <div
      v-if="$slots.leftIcon"
      class="pointer-events-none absolute inset-y-0 start-0 flex items-center ps-3.5"
    >
      <slot name="leftIcon" />
    </div>

    <slot />

    <div
      v-if="$slots.rightIcon"
      class="pointer-events-none absolute inset-y-0 end-0 flex items-center pe-3.5"
    >
      <slot name="rightIcon" />
    </div>
  </div>
</template>
