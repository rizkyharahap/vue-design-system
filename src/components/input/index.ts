import type { InputHTMLAttributes } from "vue";

export const baseClass = `block w-full rounded-lg border disabled:pointer-events-none disabled:opacity-60`;

export const inputVariants = {
  default: `border-gray-300 bg-gray-50 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500`,
  success: `border-green-500 bg-green-50 text-green-900 placeholder-green-700 focus:border-green-500 focus:ring-green-500 dark:border-green-500 dark:bg-gray-700 dark:text-green-400 dark:placeholder-green-500`,
  error: `border-red-500 bg-red-50 text-red-900 placeholder-red-700 focus:border-red-500 focus:ring-red-500 dark:border-red-500 dark:bg-gray-700 dark:text-red-500 dark:placeholder-red-500`,
};
export type InputVariants = keyof typeof inputVariants;

export const inputSizes = {
  sm: `p-2 text-xs`,
  normal: `p-2.5 text-sm`,
  lg: `p-4 text-base`,
};
export type InputSizes = keyof typeof inputSizes;

export interface InputPropsProps extends /* @vue-ignore */ InputHTMLAttributes {
  variant?: InputVariants;
  size?: InputSizes;
}

export { default as Input } from "./Input.vue";
