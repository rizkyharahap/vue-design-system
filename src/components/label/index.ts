import type { LabelHTMLAttributes } from "vue";

export const baseClass = `block mb-2 text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70`;

export const labelVariants = {
  default: `text-gray-900 dark:text-white`,
  success: `text-green-700 dark:text-green-500`,
  error: `text-red-700 dark:text-red-500`,
};
export type LabelVariants = keyof typeof labelVariants;

export interface LabelProps extends /* @vue-ignore */ LabelHTMLAttributes {
  variant?: LabelVariants;
}

export { default as Label } from "./Label.vue";
