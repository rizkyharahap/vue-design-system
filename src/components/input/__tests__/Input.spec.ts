import { shallowMount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import { Input } from "@/components/input";

describe("Input", () => {
  it("match snapshot", () => {
    const wrapper = shallowMount(Input, {
      props: { value: "input value" },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders properly", () => {
    const wrapper = shallowMount(Input, {
      props: { value: "input value" },
    });
    expect(wrapper.find("input").element.value).toBe("input value");
  });
});
