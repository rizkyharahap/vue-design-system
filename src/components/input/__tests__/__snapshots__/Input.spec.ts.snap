// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > match snapshot 1`] = `"<input type="text" id="large-input" class="p-2.5 text-sm border-gray-300 bg-gray-50 text-gray-900 focus:border-blue-500 focus:ring-blue-500 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:placeholder-gray-400 dark:focus:border-blue-500 dark:focus:ring-blue-500 block w-full rounded-lg border disabled:pointer-events-none disabled:opacity-60" value="input value">"`;
